src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IsPatternOperator.cs

```diff
@@ -15,22 +15,16 @@ internal sealed partial class LocalRewriter
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
             BoundDecisionDag decisionDag = node.GetDecisionDagForLowering(_factory.Compilation);
-            bool negated = node.IsNegated;
             BoundExpression result;
-            if (canProduceLinearSequence(decisionDag.RootNode, whenTrueLabel: node.WhenTrueLabel, whenFalseLabel: node.WhenFalseLabel))
+            if (decisionDag.RootNode is BoundLeafDecisionDagNode l && l.Label == node.WhenFalseLabel)
             {
-                // If we can build a linear test sequence `(e1 && e2 && e3)` for the dag, do so.
-                var isPatternRewriter = new IsPatternExpressionLinearLocalRewriter(node, this);
-                result = isPatternRewriter.LowerIsPatternAsLinearTestSequence(node, decisionDag, whenTrueLabel: node.WhenTrueLabel, whenFalseLabel: node.WhenFalseLabel);
-                isPatternRewriter.Free();
+                result = _factory.Literal(false);
```

Comment:
>result = _factory.Literal(false); [](http://example.com/codeflow?start=16&length=33)

Is there a targeted test covering this code path?

Reply:
There's two ways an is-expression can result in a constant:

- if the pattern is logically irrefutable (warns - caught by linear rewriter) or impossible (error - won't reach codegen)
- same as above but due to a constant input (warns for both cases)

In the latter, the linear check would catch both and produce a constant, but since we're removing the inverted one, constant false needs to be handled separately.

There's a few of tests for either scenario none of which verifies codegen except `IsNot_12`.  I'll add more in the next iteration.